.github/workflows/load_testing.yml

```
@@ -0,0 +1,31 @@
+name: Load testing
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "*" ]
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  load_testing:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Update submodules
+      run: git submodule update --init --recursive
+
+    - name: Configure CMake for building project
+      run: cmake -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DBUILD_PROJECT=ON -DBUILD_TESTS=OFF
+
+    - name: Build project
+      run: cmake --build build --config Debug -j $(nproc)
```

I think in load testing we should use release version to test the best experience. Also `--config` is redundant here as we already setup build in previous step
```suggestion
      run: cmake --build build  -j $(nproc)
```